documentation/specs/proposed/BuildCheck.md

```diff
@@ -99,7 +101,7 @@ The proposed initial configuration for those is TBD (as well based on initial te
 
 ### Live Build
 
-BuildCheck will run as part of the build and execute [inbox analyzers](#inbox-analyzers) and [custom analyzers](#acquisition-of-custom-analyzers) based on the [configuration](#configuration). Users will have an option to completely opt-out from BuildCheck to run via commandline switch.
+BuildCheck will run as part of the build and execute [inbox analyzers](#inbox-analyzers) and [custom analyzers](#acquisition-of-custom-analyzers) based on the [configuration](#configuration). Users will have an option to completely opt-out from BuildCheck to run via an MSBuild property (could be set in a project file or passed on the commandline).
```

Comment:
Property can have the disadvantage of having different values in different projects (which can get bit complicated for 'multi-project' rules like 'double writes') and disadvantage of being known later during evaluation (so prevents us from speeding up cases with evaluation analyzers enabled).

The switch can be problematic when executed from VS - but still there should be a way how to pass a global build flag (env var, etc.).

Would you see those situations as possible reasons for adjusting this direction?

Reply:
Good point about multi-project rules. Although, do you think we would be able avoid running into this situation even if we had `/analyze`? With a per-project `.editorconfig` such an analyzer can also be enabled only for a subset of projects being built.

Reply:
Yeah - it sounds like regalrdless of the opt-in/-out switch vs property we will still have same problems to solve - so going with more 'msbuild-y' way of expressing opt-in/-out might be preferable

Reply:
There is actually one scenario I'm wondering about: 

Parsing editorconfigs only when required and only once per build

Reading and parsing editorconfigs is possibly expensive operation - so it would be good to do it only once per build and share the structured representaion between the nodes. However we need to parse editorconfigs only if anlysis is requested - which we'd find only during evaluation - which is too late for exchanging node configurations.

Any thoughts on this would be very welcomed! :-)

FYI @f-alizada - for the single parsing and sharing of the structured configuration representation

Reply:
Sounds plausible